--- src/io_nix.c	2008-03-29 16:53:32.000000000 -0500
+++ src/io_nix.c.new	2009-11-24 08:36:45.000000000 -0600
@@ -192,6 +192,11 @@
 
 	addr.l2_psm = htobs(WM_OUTPUT_CHANNEL);
 
+	/* You cannot guarantee that this will automatically be set to 0 so
+	 * it is explicitly set here.
+	 */
+	addr.l2_cid = 0;
+
 	/* connect to wiimote */
 	if (connect(wm->out_sock, (struct sockaddr*)&addr, sizeof(addr)) < 0) {
 		perror("connect() output sock");
